src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,453 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 18 Mar 2020
+
+@author: Philip Winkler
+
+Copyright © 2020, Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division
+
+import glob
+import logging
+from odemis import model
+from odemis.model import Actuator, CancellableThreadPoolExecutor
+from odemis.util import to_str_escape
+import os
+import threading
+import time
+import serial
+from threading import Thread
+import copy
+from odemis.model import Axis, HwComponent
+
+BAUDRATE = 115200
+WFM_STEPSIZE = 5e-6  # m / step
```

Comment:
I'm wondering whether we need to have the accurate value in the code at all. We need to configure the stepsize (encoder counts --> motor counts) on the hardware and this value will be stored in EEPROM. In case of closed loop control, we will only pass the encoder counts and the conversion will be done internally. On the other hand, if we use open loop control, the amount of steps is approximate anyway, so an approximate stepsize might be good enough. This would save quite a lot of trouble, because we don't need to set this parameter accurately in every yaml file (which might be prone to mistakes).

Reply:
I think that for now, all the hardware will be closed-loop, so we just don't need to worry about open-loop. For closed-loop, this info will indeed probably not be needed, as it's just the encoder which returns the information.